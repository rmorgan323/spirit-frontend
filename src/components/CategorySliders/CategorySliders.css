.CategorySliders {
  width: 190px;
  margin: 10px; }
  .CategorySliders .category-title {
    height: 72px;
    background-color: #eee;
    margin-bottom: 20px;
    padding: 5px 5px;
    text-align: center;
    display: flex;
    align-items: center;
    justify-content: center;
    border-radius: 5px; }
