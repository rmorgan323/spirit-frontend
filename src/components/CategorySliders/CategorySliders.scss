.CategorySliders {
  width: 170px;
  margin: 10px;

  .category-title {
    height: 72px;
    background-color: #eee;
    margin-bottom: 20px;
    padding: 5px 2px;
    text-align: center;
  }
}