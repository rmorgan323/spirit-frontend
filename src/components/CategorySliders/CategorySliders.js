import React from 'react';
import { PropTypes } from 'prop-types';
import SliderTen from '../../containers/SliderTen/SliderTen';
import './CategorySliders.css';

const CategorySliders = ({
  title,
  slidersArray,
  databaseNamesArray,
  reminderAsterisk,
  reminder
}) => {
  const display = slidersArray.map((slider, index) => {
    return (
      <SliderTen
        sliderTitle={slider}
        databaseName={databaseNamesArray[index]}
        reminderAsterisk={reminderAsterisk ? reminderAsterisk[index] : null}
        key={index}
      />
    );
  });

  let displayTitle;

  if (title) {
    displayTitle = <h4 className="category-title">{title}</h4>;
  }

  return (
    <div className="CategorySliders">
      {displayTitle}
      {display}
      <h5 className="reminder-message">{reminder}</h5>
    </div>
  );
};

export default CategorySliders;

CategorySliders.propTypes = {
  title: PropTypes.string,
  slidersArray: PropTypes.array,
  databaseNamesArray: PropTypes.array,
  reminderAsterisk: PropTypes.array,
  reminder: PropTypes.string
};
