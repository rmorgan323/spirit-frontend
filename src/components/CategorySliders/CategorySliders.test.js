import React from 'react';
import CategorySliders from './CategorySliders';
import { shallow } from 'enzyme';
import {
  mockTitle,
  mockSlidersArray,
  mockDatabaseNamesArray,
  mockReminderAsterisk,
  mockReminder
} from '../../data/mockData/mockCategorySlidersData';

describe('CategorySliders tests', () => {
  it('Should match snapshot', () => {
    const renderedApp = shallow(
      <CategorySliders
        title={mockTitle}
        slidersArray={mockSlidersArray}
        databaseNamesArray={mockDatabaseNamesArray}
        reminderAsterisk={mockReminderAsterisk}
        reminder={mockReminder}
      />
    );

    expect(renderedApp).toMatchSnapshot();
  });

  it('Should match the snapshot if only some props are passed in', () => {
    const renderedApp = shallow(
      <CategorySliders
        slidersArray={mockSlidersArray}
        databaseNamesArray={mockDatabaseNamesArray}
      />
    );

    expect(renderedApp).toMatchSnapshot();
  });
});
