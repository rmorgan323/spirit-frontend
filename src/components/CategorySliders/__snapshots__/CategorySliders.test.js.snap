// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategorySliders tests Should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CategorySliders
    databaseNamesArray={
      Array [
        "mod_2_attention",
        "mod_2_action",
        "mod_2_affect",
        "mod_2_autonomic",
        "mod_2_arousability",
      ]
    }
    reminder="* If selected, remember to make a hypothesized arousal chart."
    reminderAsterisk={
      Array [
        null,
        "*",
        "*",
        "*",
        "*",
      ]
    }
    slidersArray={
      Array [
        "Attention",
        "Action",
        "Affect",
        "Autonomic",
        "Arousability",
      ]
    }
    title="Capacities"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h4
          className="category-title"
        >
          Capacities
        </h4>,
        Array [
          <Connect(OnClickOutside(SliderTen))
            databaseName="mod_2_attention"
            reminderAsterisk={null}
            sliderTitle="Attention"
          />,
          <Connect(OnClickOutside(SliderTen))
            databaseName="mod_2_action"
            reminderAsterisk="*"
            sliderTitle="Action"
          />,
          <Connect(OnClickOutside(SliderTen))
            databaseName="mod_2_affect"
            reminderAsterisk="*"
            sliderTitle="Affect"
          />,
          <Connect(OnClickOutside(SliderTen))
            databaseName="mod_2_autonomic"
            reminderAsterisk="*"
            sliderTitle="Autonomic"
          />,
          <Connect(OnClickOutside(SliderTen))
            databaseName="mod_2_arousability"
            reminderAsterisk="*"
            sliderTitle="Arousability"
          />,
        ],
        <h5
          className="reminder-message"
        >
          * If selected, remember to make a hypothesized arousal chart.
        </h5>,
      ],
      "className": "CategorySliders",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Capacities",
          "className": "category-title",
        },
        "ref": null,
        "rendered": "Capacities",
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "class",
        "props": Object {
          "databaseName": "mod_2_attention",
          "reminderAsterisk": null,
          "sliderTitle": "Attention",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "class",
        "props": Object {
          "databaseName": "mod_2_action",
          "reminderAsterisk": "*",
          "sliderTitle": "Action",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "class",
        "props": Object {
          "databaseName": "mod_2_affect",
          "reminderAsterisk": "*",
          "sliderTitle": "Affect",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "class",
        "props": Object {
          "databaseName": "mod_2_autonomic",
          "reminderAsterisk": "*",
          "sliderTitle": "Autonomic",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "4",
        "nodeType": "class",
        "props": Object {
          "databaseName": "mod_2_arousability",
          "reminderAsterisk": "*",
          "sliderTitle": "Arousability",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "* If selected, remember to make a hypothesized arousal chart.",
          "className": "reminder-message",
        },
        "ref": null,
        "rendered": "* If selected, remember to make a hypothesized arousal chart.",
        "type": "h5",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h4
            className="category-title"
          >
            Capacities
          </h4>,
          Array [
            <Connect(OnClickOutside(SliderTen))
              databaseName="mod_2_attention"
              reminderAsterisk={null}
              sliderTitle="Attention"
            />,
            <Connect(OnClickOutside(SliderTen))
              databaseName="mod_2_action"
              reminderAsterisk="*"
              sliderTitle="Action"
            />,
            <Connect(OnClickOutside(SliderTen))
              databaseName="mod_2_affect"
              reminderAsterisk="*"
              sliderTitle="Affect"
            />,
            <Connect(OnClickOutside(SliderTen))
              databaseName="mod_2_autonomic"
              reminderAsterisk="*"
              sliderTitle="Autonomic"
            />,
            <Connect(OnClickOutside(SliderTen))
              databaseName="mod_2_arousability"
              reminderAsterisk="*"
              sliderTitle="Arousability"
            />,
          ],
          <h5
            className="reminder-message"
          >
            * If selected, remember to make a hypothesized arousal chart.
          </h5>,
        ],
        "className": "CategorySliders",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Capacities",
            "className": "category-title",
          },
          "ref": null,
          "rendered": "Capacities",
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "class",
          "props": Object {
            "databaseName": "mod_2_attention",
            "reminderAsterisk": null,
            "sliderTitle": "Attention",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "databaseName": "mod_2_action",
            "reminderAsterisk": "*",
            "sliderTitle": "Action",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "class",
          "props": Object {
            "databaseName": "mod_2_affect",
            "reminderAsterisk": "*",
            "sliderTitle": "Affect",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "class",
          "props": Object {
            "databaseName": "mod_2_autonomic",
            "reminderAsterisk": "*",
            "sliderTitle": "Autonomic",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "4",
          "nodeType": "class",
          "props": Object {
            "databaseName": "mod_2_arousability",
            "reminderAsterisk": "*",
            "sliderTitle": "Arousability",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "* If selected, remember to make a hypothesized arousal chart.",
            "className": "reminder-message",
          },
          "ref": null,
          "rendered": "* If selected, remember to make a hypothesized arousal chart.",
          "type": "h5",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`CategorySliders tests Should match the snapshot if only some props are passed in 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CategorySliders
    databaseNamesArray={
      Array [
        "mod_2_attention",
        "mod_2_action",
        "mod_2_affect",
        "mod_2_autonomic",
        "mod_2_arousability",
      ]
    }
    slidersArray={
      Array [
        "Attention",
        "Action",
        "Affect",
        "Autonomic",
        "Arousability",
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        undefined,
        Array [
          <Connect(OnClickOutside(SliderTen))
            databaseName="mod_2_attention"
            reminderAsterisk={null}
            sliderTitle="Attention"
          />,
          <Connect(OnClickOutside(SliderTen))
            databaseName="mod_2_action"
            reminderAsterisk={null}
            sliderTitle="Action"
          />,
          <Connect(OnClickOutside(SliderTen))
            databaseName="mod_2_affect"
            reminderAsterisk={null}
            sliderTitle="Affect"
          />,
          <Connect(OnClickOutside(SliderTen))
            databaseName="mod_2_autonomic"
            reminderAsterisk={null}
            sliderTitle="Autonomic"
          />,
          <Connect(OnClickOutside(SliderTen))
            databaseName="mod_2_arousability"
            reminderAsterisk={null}
            sliderTitle="Arousability"
          />,
        ],
        <h5
          className="reminder-message"
        />,
      ],
      "className": "CategorySliders",
    },
    "ref": null,
    "rendered": Array [
      undefined,
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "class",
        "props": Object {
          "databaseName": "mod_2_attention",
          "reminderAsterisk": null,
          "sliderTitle": "Attention",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "class",
        "props": Object {
          "databaseName": "mod_2_action",
          "reminderAsterisk": null,
          "sliderTitle": "Action",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "class",
        "props": Object {
          "databaseName": "mod_2_affect",
          "reminderAsterisk": null,
          "sliderTitle": "Affect",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "class",
        "props": Object {
          "databaseName": "mod_2_autonomic",
          "reminderAsterisk": null,
          "sliderTitle": "Autonomic",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "4",
        "nodeType": "class",
        "props": Object {
          "databaseName": "mod_2_arousability",
          "reminderAsterisk": null,
          "sliderTitle": "Arousability",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "reminder-message",
        },
        "ref": null,
        "rendered": null,
        "type": "h5",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          undefined,
          Array [
            <Connect(OnClickOutside(SliderTen))
              databaseName="mod_2_attention"
              reminderAsterisk={null}
              sliderTitle="Attention"
            />,
            <Connect(OnClickOutside(SliderTen))
              databaseName="mod_2_action"
              reminderAsterisk={null}
              sliderTitle="Action"
            />,
            <Connect(OnClickOutside(SliderTen))
              databaseName="mod_2_affect"
              reminderAsterisk={null}
              sliderTitle="Affect"
            />,
            <Connect(OnClickOutside(SliderTen))
              databaseName="mod_2_autonomic"
              reminderAsterisk={null}
              sliderTitle="Autonomic"
            />,
            <Connect(OnClickOutside(SliderTen))
              databaseName="mod_2_arousability"
              reminderAsterisk={null}
              sliderTitle="Arousability"
            />,
          ],
          <h5
            className="reminder-message"
          />,
        ],
        "className": "CategorySliders",
      },
      "ref": null,
      "rendered": Array [
        undefined,
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "class",
          "props": Object {
            "databaseName": "mod_2_attention",
            "reminderAsterisk": null,
            "sliderTitle": "Attention",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "databaseName": "mod_2_action",
            "reminderAsterisk": null,
            "sliderTitle": "Action",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "class",
          "props": Object {
            "databaseName": "mod_2_affect",
            "reminderAsterisk": null,
            "sliderTitle": "Affect",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "class",
          "props": Object {
            "databaseName": "mod_2_autonomic",
            "reminderAsterisk": null,
            "sliderTitle": "Autonomic",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "4",
          "nodeType": "class",
          "props": Object {
            "databaseName": "mod_2_arousability",
            "reminderAsterisk": null,
            "sliderTitle": "Arousability",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "reminder-message",
          },
          "ref": null,
          "rendered": null,
          "type": "h5",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
